pkg/repository/provider/unified_repo.go

```diff
@@ -185,6 +283,22 @@ func (urp *unifiedRepoProvider) getRepoOption(param RepoParam) (udmrepo.RepoOpti
 	return repoOption, nil
 }
 
+func (urp *unifiedRepoProvider) getRepoOptionForOpen(param RepoParam, purpose string) (udmrepo.RepoOptions, error) {
```

Comment:
duplicate code with `getRepoOption`? 
Looks like it would be better to apply _Options Pattern_ here to construct the repoOptions:
https://golang.cafe/blog/golang-functional-options-pattern.html

Reply:
Done, followed the Options Pattern.